examples/ssh/ssh.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+import asyncio
+import pathlib
+import sys
+from typing import Optional
+
+from datetime import datetime, timedelta
+
+from yapapi import (
+    Golem,
+    NoPaymentAccountError,
+    __version__ as yapapi_version,
+    WorkContext,
+    windows_event_loop_fix,
+)
+from yapapi.log import enable_default_logger, log_summary, log_event_repr  # noqa
+from yapapi.payload import vm
+from yapapi.rest.net import Network, Node
+from yapapi.services import Service, Cluster
+
+examples_dir = pathlib.Path(__file__).resolve().parent.parent
+sys.path.append(str(examples_dir))
+
+from utils import (
+    build_parser,
+    TEXT_COLOR_CYAN,
+    TEXT_COLOR_DEFAULT,
+    TEXT_COLOR_RED,
+    TEXT_COLOR_YELLOW,
+)
+
+
+class SshService(Service):
+    def __init__(self, cluster: "Cluster", ctx: WorkContext, network: Network):
+        super().__init__(cluster, ctx)
+        self._network: Network = network
+        self._node: Optional[Node] = None
+
+    @staticmethod
+    async def get_payload():
+        return await vm.repo(
+            image_hash="23145d371090cfdac5715240141ad0735e1ce507b11ae2beef6d597f",
+            min_mem_gib=0.5,
+            min_storage_gib=2.0,
+        )
+
+    async def start(self):
+        self._node = await self._network.add_node(self.provider_id)
+        self._ctx.deploy(**self._node.get_deploy_args())
+        self._ctx.start()
+        yield self._ctx.commit()
+
+    async def run(self):
+        ip = self._node.ip
+        net = self._network.network_id
+        app_key = self.cluster._engine._api_config.app_key
+
+        print(
+            f"""Connect with:
+{TEXT_COLOR_CYAN}
+ssh -o ProxyCommand='websocat asyncstdio: ws://127.0.0.1:7465/net-api/v1/net/{net}/tcp/{ip}/22 --binary -H=Authorization:"Bearer {app_key}"' root@{ip}
+{TEXT_COLOR_DEFAULT}"""
+        )
+
+        self._ctx.run("/bin/bash", "-c", "syslogd")
+        self._ctx.run("/bin/bash", "-c", "ssh-keygen -A")
+        self._ctx.run("/bin/bash", "-c", "/usr/sbin/sshd")
+        self._ctx.run("/bin/bash", "-c", "sleep 3600")
+        yield self._ctx.commit(timeout=timedelta(minutes=30))
+
+
+async def main(subnet_tag, driver=None, network=None):
+    # By passing `event_consumer=log_summary()` we enable summary logging.
+    # See the documentation of the `yapapi.log` module on how to set
+    # the level of detail and format of the logged information.
+    async with Golem(
+        budget=1.0,
+        subnet_tag=subnet_tag,
+        driver=driver,
+        network=network,
```

Comment:
I think we have a naming problem here and I'm not sure how to solve it.
There is `Golem.network` but also `golem.create_network` and those are totally different networks.

Reply:
https://github.com/golemfactory/yapapi/issues/619